File: 1137-n-th-tribonacci-number/1137-n-th-tribonacci-number.cpp
class Solution {
public:
    int tribonacci(int n) {
        if(n <= 2){
            return n == 0 ? 0 : 1;
        }
        
        int prev1 = 0, prev2 = 1, prev3 = 1, next;
        n -= 2;
        while(n--){
            next = prev1 + prev2 + prev3;
            prev1 = prev2;
            prev2 = prev3;
            prev3 = next;
        }
        
        return next;
    }
};

File: 1137-n-th-tribonacci-number/1137-n-th-tribonacci-number.java
class Solution {
    public int tribonacci(int n) {
        if(n < 2) {
            return n;
        }
        
        int[] tribonacciNumbers = new int[n+1];
        
        tribonacciNumbers[0] = 0;
        tribonacciNumbers[1] = 1;
        tribonacciNumbers[2] = 1;
        
        for(int i = 3; i <= n; i++) {
            tribonacciNumbers[i] = tribonacciNumbers[i-3] + 
                                   tribonacciNumbers[i-2] +
                                   tribonacciNumbers[i-1];
        }
        
        return tribonacciNumbers[n];
    }
}